packages/itwin/property-grid/src/PropertyGridUiItemsProvider.tsx

```diff
@@ -73,8 +75,17 @@ export class PropertyGridUiItemsProvider implements UiItemsProvider {
   }
 }
 
+/**
+ * Props for creating `PropertyGridWidget`.
+ * @beta
+ */
+export interface PropertyGridWidgetProps extends PropertyGridComponentProps {
+  /** Predicate indicating if the widget should be shown for the current selection set. */
+  shouldShow?: (selection: Readonly<KeySet>) => boolean
+}
+
 /** Component that renders `PropertyGridComponent` an hides/shows widget based on `UnifiedSelection`. */
-function PropertyGridWidget(props: PropertyGridComponentProps) {
+function PropertyGridWidget(props: PropertyGridWidgetProps) {
```

Comment:
Need to destructure props and take out `shouldShow` prop to avoid passing it down to `PropertyGridComponent`
```suggestion
function PropertyGridWidget({ shouldShow, ...props }: PropertyGridWidgetProps) {
```